client/src/remote/Config.js

```diff
@@ -91,6 +87,21 @@ export default class Config {
     return value;
   }
 
+  /**
+   * Get whole configuration for file.
+   *
+   * @param {File} file
+   *
+   * @returns {Promise<*>}
+   */
+  async getConfigForFile(file) {
```

Comment:
Couldn't we simplify this to `getForFile(file: Object, key?: string, defaultValue?: any)`? If no key is provided the whole config is returned. Otherwise the naming is a bit confusing.

Reply:
Oh OK you just extracted the method. Isn't it internal in that case?